File: sorting/bubble_sort.js
function bubbleSort(nums) {
  while(true) {
    let flag = false;
    for(let i=0; i<nums.length-1; i++) {
      if(nums[i] > nums[i+1]) {
        flag = true;
        let tmp = nums[i];
        nums[i] = nums[i+1];
        nums[i+1] = tmp;
      }
    }
    if(!flag) break;
  }
  return nums;
}

console.log(bubbleSort([3,6,2,1,8,4]));

File: old/basic/sorting/bubble_sort.py
class Solution:
  def bubble_sort(self, nums):
    swap = True
    while swap != False:
      tmp = 0
      for i in range(1, len(nums)):
        one, two = nums[i-1], nums[i]
        if one > two:
          nums[i-1], nums[i] = nums[i], nums[i-1]
          tmp += 1
      if tmp == 0:
        swap = False
    return nums

print(Solution().bubble_sort([4,3,1,5,2]))